src/libmongoc/tests/test-mongoc-client-side-encryption.c

```diff
@@ -3706,6 +3706,714 @@ explicit_encryption_destroy (ee_fixture *eef)
    bson_free (eef);
 }
 
+/* ree_fixture is a fixture for the Range Explicit Encryption prose test. */
+typedef struct {
+   bson_value_t key1ID;
+   mongoc_client_t *keyVaultClient;
+   mongoc_client_encryption_t *clientEncryption;
+   mongoc_client_t *encryptedClient;
+   mongoc_collection_t *encryptedColl;
+   // typeStr is DoublePrecision, DoubleNoPrecision, Date, Int, or Long
+   const char *typeStr;
+   char *fieldName;
+   bson_value_t zero;
+   bson_value_t six;
+   bson_value_t thirty;
+   bson_value_t twoHundred;
+   bson_value_t twoHundredOne;
+   mongoc_client_encryption_encrypt_range_opts_t *ro;
+} ree_fixture;
+
+static ree_fixture *
+range_explicit_encryption_setup (const char *typeStr)
+{
+   ree_fixture *reef = (ree_fixture *) bson_malloc0 (sizeof (ree_fixture));
```

Comment:
In C, casting from `void *` is not necessary, so I suggest that we don't cast on malloc. Reasoning can be found here: https://stackoverflow.com/a/605858